firebase/test/core/domain/use-cases/sync-collections.spec.ts

```diff
@@ -0,0 +1,220 @@
+import * as sinon from "sinon";
+import * as assert from "assert";
+import * as collection from "#domain/models/collection";
+import * as memo from "#domain/models/memo";
+import { LocalCollectionsRepository } from "#data/repositories/local-collections-repository";
+import { StoredCollectionsRepository } from "#data/repositories/stored-collections-repository";
+import { MemosRepository } from "#data/repositories/memos-repository";
+import { GitRepository } from "#data/repositories/git-repository";
+import createSinonStub from "#test/sinon-stub";
+import { SyncCollectionsUseCase } from "#domain/use-cases/collections/sync-collections";
+import { newRawLocalCollection } from "#test/core/data/schemas/collections-fakes";
+
+describe("SyncCollectionsUseCase", () => {
+  let sandbox: sinon.SinonSandbox;
+  let localCollectionsRepoStub: sinon.SinonStubbedInstance<LocalCollectionsRepository>;
+  let storedCollectionsRepoStub: sinon.SinonStubbedInstance<StoredCollectionsRepository>;
+  let memosRepoStub: sinon.SinonStubbedInstance<MemosRepository>;
+  let gitRepoStub: sinon.SinonStubbedInstance<GitRepository>;
+  let syncCollectionsUseCase: SyncCollectionsUseCase;
+
+  let validateLocalCollectionStub: sinon.SinonStub;
+  let validateStoredCollectionStub: sinon.SinonStub;
+  let validateMemoStub: sinon.SinonStub;
+
+  before(() => {
+    sandbox = sinon.createSandbox();
+
+    const localCollections = (localCollectionsRepoStub = createSinonStub(LocalCollectionsRepository, sandbox));
+    const storedCollections = (storedCollectionsRepoStub = createSinonStub(StoredCollectionsRepository, sandbox));
+    const memosRepo = (memosRepoStub = createSinonStub(MemosRepository, sandbox));
+    const gitRepo = (gitRepoStub = createSinonStub(GitRepository, sandbox));
+
+    syncCollectionsUseCase = new SyncCollectionsUseCase(localCollections, storedCollections, memosRepo, gitRepo);
+  });
+
+  beforeEach(() => {
+    validateLocalCollectionStub = sandbox.stub(collection, "validateLocalCollection");
+    validateStoredCollectionStub = sandbox.stub(collection, "validateStoredCollection");
+    validateMemoStub = sandbox.stub(memo, "validateMemo");
```

Comment:
These stubs are typed. Typing anything else than the correct method name will break when trying to compile